src/parser/mage/fire/modules/features/HotStreakWastedCrits.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import HIT_TYPES from 'game/HIT_TYPES';
+import EnemyInstances, { encodeTargetString } from 'parser/shared/modules/EnemyInstances';
+
+const debug = false;
+
+const PROC_WINDOW_MS = 200;
+
+const HOT_STREAK_CONTRIBUTORS = [
+  SPELLS.FIREBALL.id,
+  SPELLS.PYROBLAST.id,
+  SPELLS.FIRE_BLAST.id,
+  SPELLS.SCORCH.id,
+  SPELLS.PHOENIX_FLAMES_TALENT.id,
+];
+
+class HotStreakWastedCrits extends Analyzer {
+  static dependencies = {
+    enemies: EnemyInstances,
+  };
+
+  lastCastEvent = 0;
+  wastedCrits = 0;
+  hasPyromaniacProc = false;
+
+  constructor(...args) {
+    super(...args);
+    this.hasPyromaniac = this.selectedCombatant.hasTalent(SPELLS.PYROMANIAC_TALENT.id);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER), this._onCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this._onDamage);
+    this.addEventListener(Events.applybuff.to(SELECTED_PLAYER).spell(SPELLS.HOT_STREAK), this.checkForPyromaniacProc);
+
+  }
+
+  //When a spell that contributes towards Hot Streak is cast, get the event info to use for excluding the cleaves from Phoenix Flames on the damage event
+  _onCast(event) {
+    if (!HOT_STREAK_CONTRIBUTORS.includes(event.ability.guid) && event.ability.guid !== SPELLS.FLAMESTRIKE.id) {
+      return;
+    }
+    this.lastCastEvent = event;
+    this.pyromaniacProc = false;
+  }
+
+  //When a spell that contributes towards Hot Streak crits the target while Hot Streak is active, count it as a wasted crit.
+  //Excludes the cleave from Phoenix Flames (the cleave doesnt contribute towards Hot Streak) and excludes crits immediately after Pyromaniac procs, cause the player cant do anything to prevent that.
+  _onDamage(event) {
+    const spellId = event.ability.guid;
+    const castTarget = encodeTargetString(this.lastCastEvent.targetID, event.targetInstance);
+    const damageTarget = encodeTargetString(event.targetID, event.targetInstance);
+    if (!HOT_STREAK_CONTRIBUTORS.includes(spellId) || event.hitType !== HIT_TYPES.CRIT || !this.selectedCombatant.hasBuff(SPELLS.HOT_STREAK.id) || (spellId === SPELLS.PHOENIX_FLAMES_TALENT.id && castTarget !== damageTarget)) {
+      return;
+    }
+
+    if (this.hasPyromaniacProc) {
+      debug && this.log("Wasted Crit Ignored");
+    } else if (this.selectedCombatant.hasBuff(SPELLS.HOT_STREAK.id,null,-50)) {
```

Comment:
You check for .hasBuff(HOT_STREAK) just a few lines up -- maybe remove this check and move the -50ms buffer to the previous check on L55?

Reply:
this was another thing i remember adding but i dont remember why (and was in the old module so i carried it over) ill dig through my prs maybe i can find it.

Reply:
nvm i just re-read your comment. i thought you were saying to remove the -50 buffer, but you were saying to move it to the other check. yea ill do that.